src/EditorFeatures/Core/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregateTagger.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+
+    internal abstract partial class AbstractDiagnosticsTaggerProvider<TTag> where TTag : ITag
+    {
+        /// <summary>
+        /// Simple tagger that aggregates the underlying syntax/semantic compiler/analyzer taggers and presents them as
+        /// a single event source and source of tags.
+        /// </summary>
+        private sealed class AggregateTagger : ITagger<TTag>, IDisposable
+        {
+            private readonly AbstractDiagnosticsTaggerProvider<TTag> _taggerProvider;
+            private readonly ImmutableArray<ITagger<TTag>> _taggers;
+
+            public AggregateTagger(
+                AbstractDiagnosticsTaggerProvider<TTag> taggerProvider,
+                ImmutableArray<ITagger<TTag>> taggers)
+            {
+                _taggerProvider = taggerProvider;
+                _taggers = taggers;
+            }
+
+            public void Dispose()
+            {
+                foreach (var tagger in _taggers)
+                    (tagger as IDisposable)?.Dispose();
+            }
+
+            public event EventHandler<SnapshotSpanEventArgs> TagsChanged
+            {
+                add
+                {
+                    foreach (var tagger in _taggers)
+                        tagger.TagsChanged += value;
+                }
+
+                remove
+                {
+                    foreach (var tagger in _taggers)
+                        tagger.TagsChanged -= value;
+                }
+            }
```

Comment:
hooking up events to the aggregate tagger just delegates to hte individual taggers.